motr/io_req_fop.c

```diff
@@ -305,9 +305,13 @@ static void io_bottom_half(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	gen_rep = m0_fop_data(m0_rpc_item_to_fop(reply_item));
 	rw_reply = io_rw_rep_get(reply_fop);
 
-	/* Copy attributes to client if reply received from read operation */
-	if (m0_is_read_rep(reply_fop) && op->op_code ==
-	    M0_OC_READ && m0__obj_is_di_enabled(ioo)) {
+	/*
+	 * Copy attributes to client if reply received from read operation
+	 * Skipping attribute_copy() for degraded read and for read verify mode.
+	 */
+	if (m0_is_read_rep(reply_fop) && op->op_code == M0_OC_READ &&
+	    m0__obj_is_di_enabled(ioo) && !ioo->ioo_dgmode_io_sent &&
```

Comment:
Can we move all this logic to m0__obj_is_di_enabled() ?

Reply:
I believe we are skipping attribute_copy() is required for DI only and as of now we are not supporting DI in case of  degraded read and read_verify for parity, But in future we may support DI (I am not sure is it feasible  or not ) in that case it is better to have it separate. 
Also is_dgmode and is_read_verify doesn't seems to be relevant to DI, so better to keep it separate.

Reply:
Done.